src/main/java/bio/terra/janitor/service/cleanup/flight/GoogleBucketCleanupStep.java

```diff
@@ -0,0 +1,74 @@
+package bio.terra.janitor.service.cleanup.flight;
+
+import bio.terra.cloudres.common.ClientConfig;
+import bio.terra.cloudres.google.storage.StorageCow;
+import bio.terra.generated.model.CloudResourceUid;
+import bio.terra.janitor.db.JanitorDao;
+import bio.terra.janitor.db.TrackedResource;
+import bio.terra.janitor.db.TrackedResourceId;
+import bio.terra.janitor.db.TrackedResourceState;
+import bio.terra.janitor.service.cleanup.CleanupParams;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import com.google.cloud.storage.StorageException;
+import com.google.cloud.storage.StorageOptions;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Step to cleanup Google Bucket resource. */
+public class GoogleBucketCleanupStep implements Step {
+  private Logger logger = LoggerFactory.getLogger(GoogleBucketCleanupStep.class);
+
+  private final StorageCow storageCow;
+  private final JanitorDao janitorDao;
+
+  public GoogleBucketCleanupStep(ClientConfig clientConfig, JanitorDao janitorDao) {
+    this.janitorDao = janitorDao;
+    // Janitor only uses CRL Cows to delete resources. Cleanup is not needed.
+    this.storageCow = new StorageCow(clientConfig, StorageOptions.getDefaultInstance());
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext) {
+    // TODO(yonghao): Set bucket lifetime to 0 and wait once Stairway supports waiting. This is more
+    // robust for buckets
+    //  with many or very large objects.
```

Comment:
even before we have Stairway waiting, we still need to delete objects within a bucket to delete a bucket.
Fine to `TODO(PF-29): Iterate through and delete all objects within the bucket.`, but PF-29 isn't done until we can delete a bucket that has an object in it.
Also move these TODO comments to the storageCow.delete, instead of all the way up here.